dbms/src/Common/MPMCQueue.h

```diff
@@ -427,6 +415,9 @@ class MPMCQueue
         std::unique_lock lock(mu);
         for (; read_pos < write_pos; ++read_pos)
             destruct(getObj(read_pos));
+
+        read_pos = 0;
```

Comment:
what's the purpose of this?

Reply:
For safety, I worry about using `drain` without `finish` or `cancel` in future code.